src/utils/hooks/removeAccountDependenciesHooks.ts

```diff
@@ -1,9 +1,34 @@
-import { Account } from "../../types/Account";
+import { useImplicitAccounts } from "./getAccountDataHooks";
+import { useMultisigAccounts } from "./multisigHooks";
+import { Account, ImplicitAccount } from "../../types/Account";
 import { useRemovePeersByAccounts } from "../beacon/beacon";
 import { useAppDispatch } from "../redux/hooks";
 import { assetsSlice } from "../redux/slices/assetsSlice";
 import { batchesSlice } from "../redux/slices/batches";
 import { beaconSlice } from "../redux/slices/beaconSlice";
+import { multisigsSlice } from "../redux/slices/multisigsSlice";
+
+/**
+ * Hook for cleaning up data related to deleted accounts.
+ *
+ * Removes
+ *   - deleted accounts dependencies
+ *   - obsolete multisigs
+ *   - obsolete multisigs' dependencies.
+ */
+export const useRemoveDependenciesAndMultisigs = () => {
+  const dispatch = useAppDispatch();
+  const getMultisigsToRemove = useGetMultisigsToRemove();
+  const removeAccountsDependencies = useRemoveAccountsDependencies();
+
+  return (deletedAccounts: ImplicitAccount[]) => {
```

Comment:
these aren't yet removed. maybe `accountsToRemove`? 
same for `useGetMultisigsToRemove`